cmd/outback/service_env_add.go

```diff
@@ -43,13 +43,14 @@ func addEnvVar(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	updatedDefinition, err := updateTaskDefinition(t, flagServiceAddEnvVars)
+	parsedEnvVars, err := stringsToKeyValue(flagServiceAddEnvVars)
 
 	if err != nil {
 		return err
 	}
+	updatedDefinition := u.UpdateContainerDefinitionEnvVars(*t, parsedEnvVars, cfg.Repo)
 
-	registeredDefinition, err := u.RegisterTaskDefinitionWithEnvVars(updatedDefinition)
+	registeredDefinition, err := u.RegisterTaskDefinitionWithEnvVars(&updatedDefinition)
```

Comment:
Just curious why do you need to change this to use the & "memory address" as the input now?

Reply:
The AWS SDK expects the task definition arg as a pointer so have to convert `updatedDefinition` before passing it to `u.RegisterTaskDefinitionWithEnvVars`.

Previously this used an old `updateTaskDefinition` function that always returned a pointer so there was no need to generate the reference  With this update now we're using `u.RegisterTaskDefinitionWithEnvVars` which returns the full value and requires the conversion to a pointer.

Good point though. I think this highlights that the return value semantics for a lot of the functions in the Outback struct are not really consistent. We should probably normalize them all to return a pointer or the raw value in the future.